src/render-html/js.js

```diff
@@ -102,6 +102,17 @@ const handleLongPress = e => {
   if (!lastTouchEventTimestamp || Date.now() - lastTouchEventTimestamp < 500) return;
 
   lastTouchEventTimestamp = 0;
+
+  if (isTargetMessageContent(e.target) || e.target.matches('.header')) {
```

Comment:
I wonder why you are not just going with the `.message` matching. So what if we are capturing the avatar. Will simplify stuff much more.

Reply:
The event is called with the target which is at bottom in the tree. 
```
<div>
<span> Some text </span>
</div>
```

So when user touches, target in the event will be of `span` not `div`.

In our case it is of the elements in the message content (received from the server). So I am iterating till the parent of the message content.

For headers:
* private headers: there is only one target (div)
* topic headers: here too only one target (div)
* headers in home view: here we need to get message id from parent node (div). Otherwise we can need to include messageId in the `stream-text` & `topic-text` too, which will be duplicity.

Reply:
Then you will use getMessageNode to bubble up and get the `div` and then do a `getAttribute('data-msg-id')` on it.